File: src/test/java/tests/LoginTest.java
package tests;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.Assert;
import org.testng.annotations.*;
import Pages.Login_page;
import Pages.DashboardPage;

public class LoginTest {
    WebDriver driver;
    Login_page login;
    DashboardPage dashboard;

    @BeforeMethod
    public void setup() {
        WebDriverManager.chromedriver().setup();
        driver = new ChromeDriver();
        driver.get("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login");
        driver.manage().window().maximize();
        login = new Login_page(driver);
        dashboard = new DashboardPage(driver);
    }

    @Test
    public void testValidLogin() {
        login.enterUsername("Admin");
        login.enterPassword("admin123");
        login.clickLogin();
        
        Assert.assertTrue(driver.getCurrentUrl().contains("dashboard"));
        dashboard.logout();
    }

    @AfterMethod
    public void tearDown() {
        driver.quit();
    }
}

File: tests/logintest.py
import pytest
from selenium import webdriver
from pages.login_page import LoginPage
from pages.dashboard import DashboardPage

@pytest.fixture
def setup():
    driver = webdriver.Chrome()
    driver.get("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login")
    driver.maximize_window()
    yield driver
    driver.quit()

def test_valid_login(setup):
    driver = setup
    login = LoginPage(driver)
    dashboard = DashboardPage(driver)

    login.enter_username("Admin")
    login.enter_password("admin123")
    login.click_login()

    assert "dashboard" in driver.current_url.lower()
    dashboard.logout()
